code/modules/station_goals/bsa.dm

```diff
@@ -380,31 +394,19 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 		if("fire")
 			fire(usr)
 			. = TRUE
-		if("recalibrate")
-			calibrate(usr)
+		if("set_target")
+			target = locate(params["target"])
 			. = TRUE
 	if(.)
 		update_icon()
 
-/obj/machinery/computer/bsa_control/proc/calibrate(mob/user)
-	if(!GLOB.bsa_unlock)
-		return
+/obj/machinery/computer/bsa_control/proc/get_available_targets()
 	var/list/targets = list()
 	// Find all active GPS
 	for(var/datum/component/gps/G in GLOB.GPS_list)
 		if(G.tracking)
-			targets[G.gpstag] = G
-
-	if(area_aim)
-		targets += GLOB.teleportlocs
-	var/victim = tgui_input_list(user, "Select target", "Artillery Targeting", targets)
-	if(isnull(victim))
-		return
-	if(isnull(targets[victim]))
-		return
-	target = targets[victim]
-	var/datum/component/gps/log_target = target
-	log_game("[key_name(user)] has aimed the bluespace artillery strike (BSA) at [get_area_name(log_target.parent)].")
+			targets[REF(G)] = G.gpstag
```

Comment:
```suggestion
			targets[FAST_REF(G)] = G.gpstag
```
We do not use REF. We use FAST_REF instead.

Reply:
having issues with FAST_REF

Reply:
okay fixed with help of @HowToLoLu